File: Ejercicios PLAYWRIGHT/Plantilla Playwright/src/pages/DespegarFlightInfo.ts
import { Locator, Page } from '@playwright/test'

export class DespegarFlightInfo{

    readonly btnAplicar: Locator;
    readonly btnVuelta: Locator;
    readonly btnPasajeros: Locator;
    readonly btnAdultos: Locator;
    readonly btnMenores: Locator;
    readonly selectCiudad: Locator;
    readonly btnBuscar: Locator;

    readonly page: Page;
    constructor(page: Page){
    this.page = page;
    this.btnVuelta = page.locator('xpath=//div[@class="sbox5-floating-tooltip sbox5-floating-tooltip-opened"]//div[@data-month="2023-05"]/descendant::div[text()="30"]');
    this.btnAplicar = page.locator('xpath=//button[@class="sbox5-3-btn -primary -md"]');
    this.btnAdultos = page.locator('xpath=//div[contains(div/span/text(), "Adultos")]/descendant::*[@id="svg-plus-MLZpsEK"]');
    this.btnMenores = page.locator('xpath = //div[contains(div/span/text(), "Menores")]/descendant::*[@id="svg-plus-MLZpsEK"]');
    this.btnPasajeros = page.locator('xpath = //div[@class="sbox5-segment--2_IQ3"]/descendant::div[@class="sbox5-distributionPassengers"]');
    this.selectCiudad = page.locator('xpath = //li[@class="item -active"]');  
    this.btnBuscar = page.locator('xpath=//*[@class="btn-text"and contains(text(),"Buscar")] ');

}

    async searchReservation(){

//Ingresar info de ida 
  await this.page.locator('.login-incentive--close').click();
  await this.page.getByRole('textbox', { name: 'Ingresa desde dónde viajas' }).click();
  await this.page.getByRole('textbox', { name: 'Ingresa desde dónde viajas' }).fill('');
  await this.page.keyboard.press('Backspace');
  await this.page.keyboard.type('med');
  await this.page.waitForTimeout(1000);
  await this.selectCiudad.click();
  //await this.page.getByText('Medellín, Antioquia, Colombia', { exact: true }).click();
  await this.page.waitForTimeout(2000);
  await this.page.getByRole('textbox', { name: 'Ingresa hacia dónde viajas' }).click();
  await this.page.getByRole('textbox', { name: 'Ingresa hacia dónde viajas' }).click();
  await this.page.waitForTimeout(1000);
  await this.page.keyboard.type('orl');
  await this.page.waitForTimeout(1000);
  await this.selectCiudad.click();
  //await this.page.getByText('Orlando, Florida, Estados Unidos', { exact: true }).click();
  await this.page.waitForTimeout(2000);

  //Selecciono fechas
  await this.page.getByRole('textbox', { name: 'Ida' }).click();
  await this.page.getByText('16').first().click();
  await this.page.waitForTimeout(1000);
  await this.page.getByRole('textbox', { name: 'Vuelta' }).click();
  await this.page.waitForTimeout(1000);
  await this.page.locator('.calendar-arrow-right').first().click();
  await this.page.locator('.calendar-arrow-right').first().click();
  await this.page.waitForTimeout(1000);
  await this.btnVuelta.click();
  await this.page.waitForTimeout(1000);
  await this.btnAplicar.click();
  await this.page.waitForTimeout(1000);

  //Selecciono pasajeros

  await this.btnPasajeros.click();
  await this.page.waitForTimeout(1000);
  await this.btnAdultos.click();
  await this.page.waitForTimeout(500);
  
  await this.page.locator('a').filter({ hasText: 'Aplicar' }).click();
  
  await this.btnBuscar.click();
 

    }

}

File: Ejercicios SCREENPLAY/DESPEGARSCREENPLAY/src/main/java/ui/DespegarFlightInfo.java
package ui;

import net.serenitybdd.core.pages.PageObject;
import net.serenitybdd.screenplay.targets.Target;

public class DespegarFlightInfo extends PageObject{
	public static Target txtOrigen = Target.the("Origen").locatedBy("//div[@class='sbox5-segment--2_IQ3' and span[contains(text(), 'Tramo 1')]]/descendant::input[@placeholder='Ingresa desde dónde viajas']");
	public static Target txtDestino = Target.the("Destino").locatedBy("//div[@class='sbox5-segment--2_IQ3' and span[contains(text(), 'Tramo 1')]]/descendant::input[@placeholder='Ingresa hacia dónde viajas']");
	public static Target btnDestino = Target.the("Destino").locatedBy("//li[@class='item -active']");
	public static Target txtIda = Target.the("Ida").locatedBy("//div[@class='sbox5-segment--2_IQ3' and span[contains(text(), 'Tramo 1')]]/descendant::div[@class='sbox5-dates-input1']");
    public static Target txtVuelta = Target.the("Vuelta").locatedBy("//div[@class='sbox5-segment--2_IQ3' and span[contains(text(), 'Tramo 1')]]/descendant::input[@placeholder='Vuelta']");
    public static Target btnIda = Target.the("Boton Ida").locatedBy("//div[@class='sbox5-floating-tooltip sbox5-floating-tooltip-opened']//div[@data-month='2023-03']/descendant::div[@class='sbox5-monthgrid-datenumber-number' and text()='28'][1]");
    public static Target btnRight = Target.the("Right Button").locatedBy("(//*[@class='calendar-arrow-right'])[1]");
    public static Target btnVuelta = Target.the("Boton Vuelta").locatedBy("//div[@class='sbox5-floating-tooltip sbox5-floating-tooltip-opened']//div[@data-month='2023-05']/descendant::div[@class='sbox5-monthgrid-datenumber-number' and text()='30'][1]");
	public static Target btnBuscar = Target.the("Boton Buscar").locatedBy("//*[@class='btn-text' and contains(text(),'Buscar')]");
	public static Target btnCookies = Target.the("Boton Cookies").locatedBy("//*[@class='btn-text' and contains(text(), 'Entendí')]");
	public static Target selectCiudad = Target.the("Seleccionar Opcion").locatedBy("//li[@class='item -active']");
	public static Target BtnPasajeros = Target.the("Seleccionar pasajeros").locatedBy("//div[@class='sbox5-segment--2_IQ3']/descendant::div[@class='sbox5-distributionPassengers']");
	public static Target BtnAdultos = Target.the("Cantidad Adultos").locatedBy("//div[contains(div/span/text(), 'Adultos')]/descendant::*[@id='svg-plus-MLZpsEK']");
	public static Target BtnMenores = Target.the("Cantidad Menores").locatedBy("//div[contains(div/span/text(), 'Menores')]/descendant::*[@id='svg-plus-MLZpsEK']");
	public static Target btnEdad = Target.the("Edad Menor").locatedBy("//*[@class='select']");
    public static Target selectEdad = Target.the("Selecciona Edad del Menor").locatedBy("//*[@class='select-option' and contains(text(), '10 años')]");
    public static Target resultadosVuelos = Target.the("Pagina resultados check").locatedBy("//span[contains(text(),'Escalas')]");
}
